bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandleAdv.java

```diff
@@ -211,6 +211,11 @@ public String toString() {
             return;
         }
 
+        if (!waitForWritable(distributionSchedule.getWriteSet(op.getEntryId()),
+                op.getEntryId(), 0, waitForWriteSetMs)) {
```

Comment:
All we need is AQ channels, right?

Reply:
WQ. 
if AQ < WQ, 1 channel is not writable and we decide to wait for AQ only we have the following options:
-- eventually get out of direct memory error.
-- fail all writes to the blocked channel ==> frequent ensemble changes caused by slightest fluke in data transfer
Sender is not slowed down in this case and keeps on sending more.

Reply:
The reason we choose AQ < WQ is to ride through some of these issues where WQ-AQ paths to the bookie are sluggish. I believe treatment must be same if a bookie is slow/sluggish or a channel to a bookie is slow.

Reply:
This is limiting the ability to make progress and taking away the advantage of WQ<AQ for low latency tail. I think it is OK to take an ensemble change. Let me know what you think.

Reply:
@jvrao there is an option to set waitTimeoutOnBackpressureMs=0 and get this behavior (immediate fail/ensemble change). 
I'd argue that in case of spiky traffic it is ok for netty to switch channel in non-writable state and back so to avoid frequent ensemble changes I'd wait a little. Ensemble changes aren't exactly free (takes time + more data in zk)